src/main/java/bio/terra/workspace/service/datareference/model/DataReference.java

```diff
@@ -0,0 +1,64 @@
+package bio.terra.workspace.service.datareference.model;
+
+import bio.terra.workspace.generated.model.DataReferenceDescription;
+import com.google.auto.value.AutoValue;
+import java.util.UUID;
+
+/**
+ * Internal representation of an uncontrolled data reference.
+ *
+ * <p>"Uncontrolled" here means that WM does not own the lifecycle of the underlying data.
+ */
+@AutoValue
+public abstract class DataReference {
+
+  /** ID of the workspace this reference belongs to. */
+  public abstract UUID workspaceId();
+
+  /** ID of the reference itself. */
+  public abstract UUID referenceId();
+
+  /** Name of the reference. Names are unique per workspace, per reference type. */
+  public abstract String name();
+
+  /** Type of this data reference. */
+  public abstract DataReferenceType referenceType();
+
+  /** Instructions for how to clone this reference (if at all). */
+  public abstract CloningInstructions cloningInstructions();
+
+  /** The actual object being referenced. */
+  public abstract ReferenceObject referenceObject();
+
+  /** Convenience method for translating to an API model DataReferenceDescription object. */
+  public DataReferenceDescription toApiModel() {
+    return new DataReferenceDescription()
+        .referenceId(referenceId())
+        .name(name())
+        .workspaceId(workspaceId())
+        .referenceType(referenceType().toApiModel())
+        .reference(((SnapshotReference) referenceObject()).toApiModel())
+        .cloningInstructions(cloningInstructions().toApiModel());
+  }
```

Comment:
the `(SnapshotReference) referenceObject()` cast will have to change once we have more models. No change necessary in this PR, just thinking out loud.

Reply:
Yep. I tried to stay away from making assumptions about how we'll handle multiple implementation types in the API model so I just left this as-is, but it'll definitely change in the future.